clients/drcachesim/tests/scattergather-aarch64.templatex

```diff
@@ -1,102 +1,129 @@
-ld1sh 32bit unscaled offset uxtw: PASS
-ld1sh 32bit unscaled offset sxtw: PASS
-ld1sh 64bit scaled offset: PASS
-ld1sh 64bit unscaled offset: PASS
-ld1sh 64bit unscaled offset Zt==Zm: PASS
-ld1w 32bit scaled offset uxtw: PASS
-ld1w 32bit scaled offset sxtw: PASS
-ld1w 32bit unpacked scaled offset uxtw: PASS
-ld1w 32bit unpacked scaled offset sxtw: PASS
-ld1w 32bit unpacked unscaled offset uxtw: PASS
-ld1w 32bit unpacked unscaled offset sxtw: PASS
-ld1w 32bit unscaled offset uxtw: PASS
-ld1w 32bit unscaled offset sxtw: PASS
-ld1w 64bit scaled offset: PASS
-ld1w 64bit unscaled offset: PASS
-ld1w 64bit unscaled offset Zt==Zm: PASS
-ld1sw 32bit unpacked scaled offset uxtw: PASS
-ld1sw 32bit unpacked scaled offset sxtw: PASS
-ld1sw 32bit unpacked unscaled offset uxtw: PASS
-ld1sw 32bit unpacked unscaled offset sxtw: PASS
-ld1sw 64bit scaled offset: PASS
-ld1sw 64bit unscaled offset: PASS
-ld1sw 64bit unscaled offset Zt==Zm: PASS
-ld1d 32bit unpacked scaled offset uxtw: PASS
-ld1d 32bit unpacked scaled offset sxtw: PASS
-ld1d 32bit unpacked unscaled offset uxtw: PASS
-ld1d 32bit unpacked unscaled offset sxtw: PASS
-ld1d 64bit scaled offset: PASS
-ld1d 64bit unscaled offset: PASS
-ld1d 64bit unscaled offset Zt==Zm: PASS
-st1b 32bit unpacked unscaled offset uxtw: PASS
-st1b 32bit unpacked unscaled offset sxtw: PASS
-st1b 32bit unscaled offset uxtw: PASS
-st1b 32bit unscaled offset sxtw: PASS
-st1b 32bit unscaled offset sxtw \(repeated offset\): PASS
-st1b 64bit unscaled offset: PASS
-st1b 64bit unscaled offset \(repeated offset\): PASS
-st1h 32bit scaled offset uxtw: PASS
-st1h 32bit scaled offset sxtw: PASS
-st1h 32bit unpacked scaled offset uxtw: PASS
-st1h 32bit unpacked scaled offset sxtw: PASS
-st1h 32bit unpacked unscaled offset uxtw: PASS
-st1h 32bit unpacked unscaled offset sxtw: PASS
-st1h 32bit unscaled offset uxtw: PASS
-st1h 32bit unscaled offset sxtw: PASS
-st1h 32bit unscaled offset sxtw: PASS
-st1h 32bit unscaled offset sxtw \(repeated offset\): PASS
-st1h 64bit scaled offset: PASS
-st1h 64bit unscaled offset: PASS
-st1h 64bit unscaled offset \(repeated offset\): PASS
-st1w 32bit scaled offset uxtw: PASS
-st1w 32bit scaled offset sxtw: PASS
-st1w 32bit unpacked scaled offset uxtw: PASS
-st1w 32bit unpacked scaled offset sxtw: PASS
-st1w 32bit unpacked unscaled offset uxtw: PASS
-st1w 32bit unpacked unscaled offset sxtw: PASS
-st1w 32bit unscaled offset uxtw: PASS
-st1w 32bit unscaled offset sxtw: PASS
-st1w 32bit unscaled offset sxtw \(repeated offset\): PASS
-st1w 64bit scaled offset: PASS
-st1w 64bit unscaled offset: PASS
-st1w 64bit unscaled offset \(repeated offset\): PASS
-st1d 32bit unpacked scaled offset uxtw: PASS
-st1d 32bit unpacked scaled offset sxtw: PASS
-st1d 32bit unpacked unscaled offset uxtw: PASS
-st1d 32bit unpacked unscaled offset sxtw: PASS
-st1d 64bit scaled offset: PASS
-st1d 64bit unscaled offset: PASS
-st1d 64bit unscaled offset \(repeated offset\): PASS
+ld1b scalar\+vector 32bit unscaled offset uxtw: PASS
```

Comment:
So this test ensures functional correctness of the app under drcachesim instrumentation. Do we have a test that verifies the generated trace?

Reply:
How would we go about verifying the trace? Is there an x86 test that does that? I created the AArch64 scatter/gather tests by creating equivalents to the existing x86 scatter/gather tests.

Reply:
We have offline-allasm-scattergather-basic-counts.templatex that verifies basic counts.

Reply:
Thanks. I'll look at adding an equivalent test in a future PR.